persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/ConversationDAOTest.kt

```diff
@@ -296,6 +317,210 @@ class ConversationDAOTest : BaseDatabaseTest() {
+        // given
+        conversationDAO.insertConversation(
+            newConversationEntity(
+                id = QualifiedIDEntity("1", "someDomain"),
+                lastReadDate = "2000-01-01T12:30:00.000Z",
+                lastModified = "2000-01-01T12:00:00.000Z"
+            )
+        )
+        conversationDAO.insertConversation(
+            newConversationEntity(
+                id = QualifiedIDEntity("2", "someDomain"),
+                lastReadDate = "2000-01-01T12:30:00.000Z",
+                lastModified = "2000-01-01T12:00:00.000Z"
+            )
+        )
+        conversationDAO.insertConversation(
+            newConversationEntity(
+                id = QualifiedIDEntity("3", "someDomain"),
+                lastReadDate = "2000-01-01T12:30:00.000Z",
+                lastModified = "2000-01-01T12:00:00.000Z"
+            )
+        )
+        conversationDAO.insertConversation(
+            newConversationEntity(
+                id = QualifiedIDEntity("3", "someDomain"),
+                lastReadDate = "2000-01-01T12:30:00.000Z",
+                lastModified = "2000-01-01T12:00:00.000Z"
+            )
+        )
+
+        // when
+        val result = conversationDAO.getUnreadConversationCount()
+
+        // then
+        assertEquals(0L, result)
+    }
+
+    @Test
+    fun givenMessagesArrivedAfterTheUserSawConversation_WhenGettingUnreadMessageCount_ThenReturnTheExpectedCount() = runTest {
+        // given
+        val conversationId = QualifiedIDEntity("1", "someDomain")
+
+        conversationDAO.insertConversation(
+            newConversationEntity(
+                id = conversationId,
+                lastReadDate = "2000-01-01T12:00:00.000Z",
+            )
+        )
+
+        userDAO.insertUser(user1)
+
+        val message = buildList {
+            // add 5 Message before the lastReadDate
+            repeat(5) {
+                add(
+                    newMessageEntity(
+                        id = it.toString(), date = "2000-01-01T12:0$it:00.000Z",
+                        conversationId = conversationId,
+                        senderUserId = user1.id,
+                    )
+                )
+            }
+            // add 5 Message past the lastReadDate
+            repeat(5) {
+                add(
+                    newMessageEntity(
+                        id = it.toString(), date = "2000-01-01T12:1$it:00.000Z",
+                        conversationId = conversationId,
+                        senderUserId = user1.id,
+                    )
+                )
+            }
+        }
+
+        messageDAO.insertMessages(message)
+
+        // when
+        val result = conversationDAO.getUnreadMessageCount(conversationId)
+
+        // then
+        assertEquals(5L, result)
+    }
+
+    @Test
+    fun givenTheConversationMessagesArrivedBeforeUserSawTheConversation_WhenGettingUnreadMessageCount_ThenReturnExpectedCount() = runTest {
```

Comment:
```suggestion
    fun givenMessagesArrivedBeforeUserSawTheConversation_whenGettingUnreadMessageCount_thenReturnZeroUnreadCount() = runTest {
```

If possible, it's better to put expectations in the test name.
So we don't need to go and actually read the test to understand what was the expected count.